docs/2025/puzzles/day05.md

```diff
@@ -2,10 +2,242 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 # Day 5: Cafeteria
 
+by [Bulby](https://github.com/TheDrawingCoder-Gamer)
+
 ## Puzzle description
 
 https://adventofcode.com/2025/day/5
 
+## Solution Summary
+
+* Parse input into list of ranges and list of ingredient IDs
+* For part 1, count the amount of ingredients that are fresh
+* For part 2, find the total size of all of the ranges combined
+  * This will require special handling to prevent overlap!
+
+## Parsing
+
+We'll need to end up doing more fancy stuff with our range type, so let's define it as a case class:
+
+```scala
+final case class LRange(start: Long, end: Long)
+```
+
+Let's define our parsed input as a tuple:
+
+```scala
+type Input = (List[LRange], List[Long])
+```
+
+Parsing isn't that bad then. Note that here I'm using `runtimeChecked`, which isn't stable yet.
+It will end up being stable when Scala 3.8.0 comes out, but if you're following along at home and don't want
+to add `@experimental` to everything, you can replace it with `: @unchecked`.
+
+```scala
+def parse(str: String): Input =
+  val Array(ranges, ings) = str.split("\n\n").runtimeChecked
+  (
+    ranges.linesIterator.map:
+      case s"$s-$e" => LRange(s.toLong, e.toLong)
+    .toList,
+    ings.linesIterator.map(_.toLong).toList
+  )
+```
+
+## Part 1
+
+We need to see if a range contains an ingredient, so let's add a method to `LRange`:
+
+```scala
+final case class LRange(start: Long, end: Long):
+  def contains(n: Long): Boolean = n >= start && n <= end
+```
+
+Then we can easily do part 1:
+
+```scala
+def part1(input: String): Long =
+  val (ranges, ingredients) = parse(input)
+  ingredients.count(ing => ranges.exists(_.contains(ing))).toLong
+```
+
+## Part 2
+
+Part 2 is a lot more complicated, but I've done [much worse](https://adventofcode.com/2021/day/22), so it wasn't too bad.
+
+My common library has implementations for ranges that can intersect and combine with each other, but I'll reimplement them here
```

Comment:
I'm a little curious, why implement a new range over NumericRange?

Reply:
NumericRange throws if your range is larger than the int limit - will add explanation to docs